haproxy/Dockerfile

```diff
@@ -0,0 +1,77 @@
+#FROM ubuntu:jammy-20230425
+FROM ubuntu:focal-20230412 AS build
+
+ENV HAPROXY_BRANCH 2.6
+ENV HAPROXY_MINOR 2.6.10
+ENV HAPROXY_SHA256 e71b2cd9ca1043345f083a5225078ccf824dced2b5779d86f11fa4e88f451773
+ENV HAPROXY_SRC_URL http://www.haproxy.org/download
+
+ENV HAPROXY_UID haproxy
+ENV HAPROXY_GID haproxy
+ENV DEBIAN_FRONTEND noninteractive
+
+# if you use ubuntu:jammy, you must use "libssl3=3.0.2-0ubuntu1.9". but it is few case.
+# We choose "ubuntu:focal" and "libssl1.1" to get a stable operation.
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends procps zlib1g libssl1.1 "libpcre2-*" liblua5.3-0 libatomic1 tar curl socat ca-certificates && \
+    apt-get install -y --no-install-recommends gcc make libc6-dev libssl-dev libpcre3-dev zlib1g-dev liblua5.3-dev && \
+    curl -sfSL "${HAPROXY_SRC_URL}/${HAPROXY_BRANCH}/src/haproxy-${HAPROXY_MINOR}.tar.gz" -o haproxy.tar.gz && \
```

Comment:
Using the `-f` option in the `curl` command ignore HTTP error.
Since the container image build should fail if an HTTP error occurs, it is better not to use the `-f` option.

Reply:
When execute "curl" without "-f" option, an HTTP server fails to deliver a document, it returns an HTML document stating so. It is bad for shell script. "curl -f" prevent curl from outputting that and return error 22.

In this case, we should use curl with "-f" option.